prog/vm/host_nexus.rb

```diff
@@ -105,6 +110,33 @@ def before_run
     }
   end
 
+  def default_boot_image_version(image_name)
+    case image_name
+    when "ubuntu-jammy"
+      Config.ubuntu_jammy_version
+    when "github-ubuntu-2204"
+      Config.github_ubuntu_2204_version
+    when "github-ubuntu-2004"
+      Config.github_ubuntu_2004_version
+    when "github-gpu-ubuntu-2204"
+      Config.github_gpu_ubuntu_2204_version
+    when "postgres-ubuntu-2204"
+      Config.postgres_ubuntu_2204_version
+    else
+      fail "Unknown boot image: #{image_name}"
+    end
+  end
+
+  label def download_boot_image
+    # check if download is complete
+    hop_prep_reboot if /image=.* version=.* downloaded/.match?(retval&.dig("msg"))
```

Comment:
retval can contain structured data, being a full JSON, I'm a bit puzzled why we have a structured format in a structured format, only to have to re-parse it. But perhaps that's merely path dependency.

Reply:
I can prepare a patch to rectify this after the fact though, unless you wanted to throw it in.

```diff
diff --git a/prog/download_boot_image.rb b/prog/download_boot_image.rb
index 849004f4..a28c0f1d 100644
--- a/prog/download_boot_image.rb
+++ b/prog/download_boot_image.rb
@@ -112,6 +112,6 @@ def blob_storage_client
       name: image_name,
       version: version
     ).update(activated_at: Time.now)
-    pop "image=#{image_name} version=#{version} downloaded"
+    pop(msg: "image downloaded", image: image_name, version: version)
   end
 end
```

and running the tests suggests that the interface width is not so large...

Reply:
...though I guess if the interface is _so_ thin I'm wondering why retval includes dynamic content.

Reply:
https://github.com/ubicloud/ubicloud/pull/1601